crates/rust-analyzer/src/hack_recover_crate_name.rs

```diff
@@ -0,0 +1,25 @@
+//! Currently cargo does not emit crate name in the `cargo test --format=json`, which needs to be changed. This
```

Comment:
Then we should get that fixed in cargo instead of creating this mess here

Reply:
I will try to fix it in cargo, By the way @ShuiRuTian did you had better solution to this in your PR?

Reply:
Not sure what here want to do.
If it's trying to find the crate of running tests, it's not a big problem in my PR. Because we do have the entire logic test tree(creates-testkinds-modules-tests) and we only allows one test to be run meanwhile, so it's easy to get the crate of the test that user clicked by calling get parent.

Reply:
Aha. This PR handles running multiple tests so it hits this problem.

It seems there is an effort in upstream (see https://github.com/rust-lang/testing-devex-team/issues/1 and https://github.com/rust-lang/rfcs/pull/3558) about reworking the test json output that will hopefully fix this problem.